### PR TITLE
added sub-national validation output script for India

### PR DESCRIPTION
**Pull request by PrantikaDas**
## :bird: Description of this PR :bird:

- The output script generates subnational spatial validation for India in comparison to India's national agriculture database

## :wrench: Checklist for PR creator :wrench:

- [X] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts

- [X] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
 
- [X] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments

- [X] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - Current develop branch default : NA
  - This PR's default :  NA

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [ ] content review done (at least 1)
- [ ] RSE review done (at least 1)

### CODE REVIEW

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+croparea_hr <- croparea_hr_share[,,]*setItems(cell_hr[,1,], dim=2, NULL)
+croparea_hr <- dimSums(croparea_hr, dim=3.2)
+
+# aggregate land from grid to state level
+mapping_ind <- mapping[ind_cells,]
+croparea_state <- toolAggregate(croparea_hr, rel=mapping_ind, from="iso_mag", to="ind")
+
+# Observed data from mrfable
+setConfig(extramappings = "mappingIndiaAPY.csv")
+ind_apy_mapping <- read.csv(system.file("extdata", "regional/mappingIndiaAPY.csv", package = "mrfable"))
```
Comment (pfuehrlich-pik) [inline scripts/output/projects/FABLE_India_subnational_validation.R:62]: Does `madrat::toolGetMapping` not work here?

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+############################# BASIC CONFIGURATION #######################################
+
+if(!exists("source_include")) {
+  
+  outputdir <- "output//v4p77/h12_2022-12-15_22.09.46"
+  
+  ###Define arguments that can be read from command line
+  readArgs("outputdir")
+}
+# hist     <- "runs/validation.mif"
```
Comment (pfuehrlich-pik) [inline scripts/output/projects/FABLE_India_subnational_validation.R:36]:
```suggestion
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+library(magpiesets)
+library(lusweave)
+
+print("Start India state level validation output runscript")
+
+############################# BASIC CONFIGURATION #######################################
+
+if(!exists("source_include")) {
+  
+  outputdir <- "output//v4p77/h12_2022-12-15_22.09.46"
```
Comment (pfuehrlich-pik) [inline scripts/output/projects/FABLE_India_subnational_validation.R:31]:
```suggestion
  outputdir <- "output/v4p77/h12_2022-12-15_22.09.46"
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+file    <- file.path(outputdir,paste0("India_subnational_validation_",format(Sys.time(), "%Y%H%M%S"),".pdf"))
+#########################################################################################
+
+print(paste0("Script started for output directory: ",outputdir))
+
+# Determine cells for India
+mapping <- toolGetMapping("scripts/npi_ndc/policies/country2cell.rds")
+mapping$iso_mag <- paste(mapping$iso,mapping$cell,sep=".")
+# load Indian states
+ind_states <- read.csv(system.file("extdata", "regional/india_state_code.csv", package = "mrfable"))
```
Comment (pfuehrlich-pik) [inline scripts/output/projects/FABLE_India_subnational_validation.R:46]: I'd suggest that we stick to camelCase in magpiemodel as well, so it is consistent with our variable naming style in R packages.
```suggestion
indStates <- read.csv(system.file("extdata", "regional/india_state_code.csv", package = "mrfable"))
```

Comment (pfuehrlich-pik) [reply scripts/output/projects/FABLE_India_subnational_validation.R:46]: No, both functions and variables should follow camelCase

File: scripts/output/projects/FABLE_India_subnational_validation.R (hunk 1 above)
Comment (mishkos) [reply scripts/output/projects/FABLE_India_subnational_validation.R:62]: it does. will change

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:45]:
```suggestion
indStates <- toolGetMapping(system.file("extdata", "regional/india_state_code.csv", package = "mrfable"))
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+ind_cells <- which(mapping$ind %in% ind_states$State_code)
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:46]:
```suggestion
indCells <- which(mapping$ind %in% indStates$State_code)
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+# Subnational validation data of specific crops
+land_hr <- read.magpie(file.path(outputdir,"cell.land_0.5.mz"))[ind_cells,,]
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:49]:
```suggestion
landHr <- read.magpie(file.path(outputdir,"cell.land_0.5.mz"))[indCells,,]
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+croparea_hr_share <- read.magpie(file.path(outputdir,"cell.croparea_0.5_share.mz"))[ind_cells,,]
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:50]:
```suggestion
cropareaHrShare <- read.magpie(file.path(outputdir,"cell.croparea_0.5_share.mz"))[indCells,,]
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+cell_hr <- dimSums(land_hr, dim=3)
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:51]:
```suggestion
cellHr <- dimSums(landHr, dim=3)
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+croparea_hr <- croparea_hr_share[,,]*setItems(cell_hr[,1,], dim=2, NULL)
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:52]:
```suggestion
cropareaHr <- cropareaHrShare[,,]*setItems(cell_hr[,1,], dim=2, NULL)
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+croparea_hr <- dimSums(croparea_hr, dim=3.2)
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:53]:
```suggestion
cropareaHr <- dimSums(cropareaHr, dim=3.2)
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+
+# aggregate land from grid to state level
+mapping_ind <- mapping[ind_cells,]
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:56]:
```suggestion
mappingInd <- mapping[indCells,]
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+croparea_state <- toolAggregate(croparea_hr, rel=mapping_ind, from="iso_mag", to="ind")
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:57]:
```suggestion
cropareaState <- toolAggregate(cropareaHr, rel=mappingInd, from="iso_mag", to="ind")
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+
+# Observed data from mrfable
+setConfig(extramappings = "mappingIndiaAPY.csv")
+ind_apy_mapping <- read.csv(system.file("extdata", "regional/mappingIndiaAPY.csv", package = "mrfable"))
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:61]:
```suggestion
indApyMapping <- toolGetMapping(system.file("extdata", "regional/mappingIndiaAPY.csv", package = "mrfable"))
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+h <- calcOutput("IndiaFoodcrop", subtype = "Area", aggregate = "Region")
+# Data cleaning:
+h <- h[ind_states$State_code,,]
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:64]:
```suggestion
h <- h[indStates$State_code,,]
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+                                              k=c("trce", "tece", "puls_pro", 
+                                                  "trce", "puls_pro", "maiz",
+                                                  "tece", "trce", "rice_pro")))
+# Aggregate to magpie crops:
+h <- toolAggregate(h, rel=mappingCropsAPY, from="APYcrop", to="k", dim=3)
+h <- add_dimension(h, dim=3.1, add="scenario", nm="historical")
+h <- add_dimension(h, dim=3.2, add="model", nm="APY")
+h <- setItems(h, dim=3.3, reportingnames(getItems(h,dim=3.3)))
+
+croparea_state <- croparea_state[,,unique(mappingCropsAPY[,"k"])]
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:81]:
```suggestion
cropareaState <- cropareaState[,,unique(mappingCropsAPY[,"k"])]
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+names(dimnames(croparea_state)) <- c("state","year","crop")
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:82]:
```suggestion
names(dimnames(cropareaState)) <- c("state","year","crop")
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+croparea_state <- add_dimension(croparea_state, dim=3.1, add="scenario", nm="")
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:83]:
```suggestion
cropareaState <- add_dimension(cropareaState, dim=3.1, add="scenario", nm="")
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+croparea_state <- add_dimension(croparea_state, dim=3.2, add="model", nm="MAgPIE")
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:84]:
```suggestion
cropareaState <- add_dimension(cropareaState, dim=3.2, add="model", nm="MAgPIE")
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 20]
```diff
@@ -0,0 +1,104 @@
...
+croparea_state <- setItems(croparea_state, dim=3.3, reportingnames(getItems(croparea_state,dim=3.3)))
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:85]:
```suggestion
cropareaState <- setItems(cropareaState, dim=3.3, reportingnames(getItems(cropareaState,dim=3.3)))
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+swlatex(sw,"\\huge")
+swlatex(sw,"\\textbf{MAgPIE state level croparea comparison for India}\\newline")
+swlatex(sw,"\\normalsize")
+swlatex(sw,"\\newline")
+swlatex(sw,"\\tableofcontents")
+
+for(k in getItems(h, dim=3.3)){
+  for(state in getItems(h, dim=1)){
+    swlatex(sw,"\\newpage")
+    swlatex(sw,paste0("\\section{Croparea ",k," in ",ind_states[which(ind_states$State_code==state),"State_name"],"}"))
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:97]:
```suggestion
    swlatex(sw,paste0("\\section{Croparea ",k," in ",indStates[which(indStates$State_code==state),"State_name"],"}"))
```

File: scripts/output/projects/FABLE_India_subnational_validation.R [hunk 22]
```diff
@@ -0,0 +1,104 @@
...
+    swfigure(sw, print, mipLineHistorical(croparea_state[state,,k],h[state,,k],ylab="Mha"))
```
Comment (mishkos) [inline scripts/output/projects/FABLE_India_subnational_validation.R:98]:
```suggestion
    swfigure(sw, print, mipLineHistorical(cropareaState[state,,k],h[state,,k],ylab="Mha"))
```

**Review by mishkos — CHANGES_REQUESTED** [review]

**Review by mishkos — APPROVED** [review]
The script is running after adopted styling changes. Looks good.

**Review by pfuehrlich-pik — APPROVED** [review]

**Pull request merged**